### PR TITLE
Refactor version filters

### PR DESCRIPTION
**Pull request by seallard**
# Summary
Refactor version filters.

## Review
- [x] code approved by HS
- [x] "Merge and deploy" approved by HS
Thanks for filling in who performed the code review and the test!

This [version](https://semver.org/) is a:
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: housekeeper/store/api/find.py [hunk 1]
```diff
@@ -53,20 +63,18 @@ def version(
         """Fetch a version from the store."""
         if version_id:
```
Comment (seallard) [inline housekeeper/store/api/find.py:64]: I don't think this pattern makes sense - to pass multiple parameters and use if-statements. We should try to split these into separate smaller functions in general, it makes the code easier to test and understand.

File: tests/store/test_version_bundle_filters.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    # THEN the version's creation date should match the given date
+    assert filtered_version.created_at == version_date
+
+def test_filter_version_by_name_and_bundle_date_with_nonexistent_data_returns_empty_query(populated_store):
+    """Test that the function returns an empty result when given a non-existent bundle name or version date."""
+
+    # GIVEN some bundle name and version date that does not exist in the store
+    nonexistent_bundle_name = "nonexistent_bundle"
+    nonexistent_version_date = dt.datetime.now()
```
Comment (henrikstranneheim) [inline tests/store/test_version_bundle_filters.py:41]: `dt.datetime.now()` this could be a `time_stamp_now` fixture

File: tests/store/test_version_filters.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
+from housekeeper.store.api.core import Store
+from housekeeper.store.models import Version
+from housekeeper.store.version_filters import filter_version_by_id
+
+
+def test_filter_version_by_id_returns_the_correct_version(populated_store: Store):
+    """Test getting version by version ID."""
+
+    # GIVEN a store with a version
+    version: Version = populated_store.session.query(Version).first()
```
Comment (henrikstranneheim) [inline tests/store/test_version_filters.py:10]:
```suggestion
    version: Version = populated_store._get_version_query.first()
```
Let not couple query to the test suit

File: tests/store/test_version_filters.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    assert version
+
+    version_id: int = version.id
+
+    # WHEN retrieving the version by ID
+    filtered_version: Version = filter_version_by_id(
+        versions=populated_store.session.query(Version),
```
Comment (henrikstranneheim) [inline tests/store/test_version_filters.py:17]: Same here

File: tests/store/test_version_filters.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    assert isinstance(filtered_version, Version)
+
+    # THEN the id should match
+    assert filtered_version.id == version_id
+
+
+def test_filter_version_by_id_with_invalid_id_returns_empty_query(populated_store: Store):
+    """Test that an empty result is returned when using an invalid version ID."""
+
+    # GIVEN a store with some versions
```
Comment (henrikstranneheim) [inline tests/store/test_version_filters.py:31]:
```suggestion
    # GIVEN a store with versions
```

File: tests/store/test_version_filters.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    versions_query = populated_store.session.query(Version)
```
Comment (henrikstranneheim) [inline tests/store/test_version_filters.py:32]: same here

File: tests/store/test_version_filters.py [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+    assert versions_query.count() > 0
+
+    # GIVEN an invalid id that doesn't exist in the store
+    invalid_version_id = versions_query.order_by(Version.id.desc()).first().id + 1
```
Comment (henrikstranneheim) [inline tests/store/test_version_filters.py:36]: Should`invalid_version_id` be a fixture for reuse and maybe it is ok to hard code the return from the `order_by` in the fixture? An invalid version id should not change much whatever we do to the rest of the code

File: housekeeper/store/api/find.py [hunk 8]
```diff
@@ -30,6 +32,14 @@ def _get_bundle_query(self) -> Query:
         """Return bundle query."""
         return self.Bundle.query
 
+    def _get_version_query(self) -> Query:
+        """Return version query."""
+        return self.Version.query
+
+    def _get_version_bundle_query(self) -> Query:
```
Comment (henrikstranneheim) [inline housekeeper/store/api/find.py:39]: Would be great to add simple unit test for these as they are orthogonal to the filters and might break when we retire alchy and/or update  sqlalchemy

**Review by henrikstranneheim — APPROVED** [review]
Looking good 👍

File: tests/store/test_version_filters.py (hunk 7 above)
Comment (seallard) [reply tests/store/test_version_filters.py:36]: Hm, invalid was a bad name. I'll keep it as it is and change the name to non_existing_id for now.

Comment (seallard) [reply tests/store/test_version_filters.py:36]: I'm not sure if it should be a fixture, it would only be guaranteed to not exist for the specific store the fixture uses.

File: housekeeper/store/api/find.py (hunk 1 above)
Comment (seallard) [reply housekeeper/store/api/find.py:64]: Saving the refactoring to a separate PR
